syntax = 'proto3';
package gooseai;
option go_package = "./;engines";

// Possible engine type
enum EngineType {
  TEXT = 0;
  PICTURE = 1;
  AUDIO = 2;
  VIDEO = 3;
  CLASSIFICATION = 4;
  STORAGE = 5;
}

enum EngineTokenizer {
  GPT2 = 0;
  PILE = 1;
}

// Engine info struct
message EngineInfo {
  string id = 1;
  string owner = 2;
  bool ready = 3;
  EngineType type = 4;
  EngineTokenizer tokenizer = 5;
  string name = 6;
  string description = 7;
}

message ListEnginesRequest {
  // Empty
}

// Engine info list
message Engines {
  repeated EngineInfo engine = 1;
}

service EnginesService {
  rpc ListEngines (ListEnginesRequest) returns (Engines) {};
}
